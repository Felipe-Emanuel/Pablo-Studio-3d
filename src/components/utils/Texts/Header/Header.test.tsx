import { render } from "@testing-library/react";
import { Header } from ".";

const config = {
  title: "Project Title",
  subtitle: "Collection",
};

describe("<Header />", () => {
  const { subtitle, title } = config;
  it("should render without errors", () => {
    render(
      <Header
        title={title}
        subtitle={subtitle}
        subtitleSize="lg"
        titleSize="2xl"
      />
    );
  });

  it("should render title and subtitle", () => {
    const { container } = render(<Header title={title} subtitle={subtitle} />);

    const h1 = container.querySelector(`h1`);
    const h2 = container.querySelector(`h2`);

    expect(h1?.textContent).toBe("Project Title");
    expect(h2?.textContent).toBe("Collection");
  });

  it("should render without title and subtitle", () => {
    //@ts-expect-error
    const { container } = render(<Header subtitleSize="lg" titleSize="2xl" />);

    expect(container).toBeInTheDocument();
  });

  it("should apply 'dark' mode ", () => {
    const { getByText } = render(
      <Header title={title} subtitle={subtitle} dark />
    );

    const h1 = getByText(title);

    expect(h1).toHaveClass("text-black");
  });
});
