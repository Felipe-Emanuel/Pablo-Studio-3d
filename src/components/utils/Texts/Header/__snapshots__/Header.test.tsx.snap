// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render title and subtitle 1`] = `
<div>
  <div
    class="flex flex-col gap-0 w-full max-w-sm leading-3"
  >
    <h1
      class="font-Roboto text-xl text-white font-bold leading-[65px]"
    >
      Project Title
    </h1>
    <h2
      class="font-Roboto text-lg text-white font-bold text-white/75"
    >
      Collection
    </h2>
  </div>
</div>
`;
